packages/api/src/promise/index.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017-2018 @polkadot/api authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { ApiPromiseInterface, QueryableStorageFunction, QueryableModuleStorage, QueryableStorage, SubmittableExtrinsics, SubmittableModuleExtrinsics, SubmittableExtrinsicFunction } from './types';
+
+import WsProvider from '@polkadot/rpc-provider/ws';
+import Rpc from '@polkadot/rpc-core/index';
+import { Extrinsics, ExtrinsicFunction } from '@polkadot/extrinsics/types';
+import { Storage } from '@polkadot/storage/types';
+import { Base } from '@polkadot/types/codec';
+
+import ApiBase from '../Base';
+import SubmittableExtrinsic from './SubmittableExtrinsic';
+import { StorageFunction } from '@polkadot/types/StorageKey';
+
+export default class ApiPromise extends ApiBase<Rpc, QueryableStorage, SubmittableExtrinsics> implements ApiPromiseInterface {
+  private _isReady: Promise<ApiPromise>;
+
+  /**
+   * @description Creates an ApiPromise instance using the supplied provider. Returns an Promise containing the actual Api instance.
+   * @param wsProvider Optional WebSocket provider that is passed to the class contructor
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * import Api from '@polkadot/api/promise';
+   *
+   * Api.create().then(async (api) => {
+   *   const timestamp = await api.st.timestamp.now();
+   *
+   *   console.log(`lastest block timestamp ${timestamp}`);
+   * });
+   * ```
+   */
+  static create (wsProvider?: WsProvider): Promise<ApiPromise> {
```

Comment:
could we update the example to show how to use the optional `wsProvider` parameter.
i think it'd be used as follows:
```
   * ```javascript
   * import Api from '@polkadot/api/promise';
   * import Ws from '@polkadot/rpc-provider/ws';
   *
   * const wsProvider = new Ws('ws://127.0.0.1:9944');
   * 
   * Api.create(wsProvider).then(async (api) => {
   *   const timestamp = await api.st.timestamp.now();
   *
   *   console.log(`lastest block timestamp ${timestamp}`);
   * });
   * ```
```

Reply:
100% right. The next PR trying to make a start at actual docs. (All these need to go into examples)